base/random.jl

```diff
@@ -253,14 +253,10 @@ rand(r::MersenneTwister, ::Type{Int128})  = reinterpret(Int128, rand(r, UInt128)
 rand{T<:Real}(r::AbstractRNG, ::Type{Complex{T}}) = complex(rand(r, T), rand(r, T))
 
 # random Char values
-# use simple rejection sampling over valid Char codepoint range
+# returns a random valid Unicode scalar value (i.e. 0 - 0xd7ff, 0xe000 - # 0x10ffff)
 function rand(r::AbstractRNG, ::Type{Char})
-    while true
-        c = rand(0x00000000:0x0010fffd)
-        if is_valid_char(c)
-            return reinterpret(Char,c)
-        end
-    end
+    c = rand(0x00000000:0x0010f7ff)
```

Comment:
I just followed what the code did before (it just had the values incorrect).  Why would the r be needed, it actually is not even used (I don't know why the function even has the argument, it seems rather misleading)